pkg/msgconv/from-whatsapp.go

```diff
@@ -120,6 +122,10 @@ type AttachmentMessageWithCaption[Integral MediaTransportContainer, Ancillary an
 
 type convertFunc func(ctx context.Context, data []byte, mimeType string) ([]byte, string, string, error)
 
+type userVisibleError struct {
+	error
+}
```

Comment:
It'd be better to just have an `Message string` inside the struct and implement the error interface yourself. That way you also avoid the linter error complaining about uppercase error messages, so you won't need the hacky capitalization afterwards